contract.xml

```diff
@@ -504,7 +492,32 @@
             <field name="act_window" ref="act_contract_review_line"/>
         </record>
 
-        <!-- Menus -->
+	<!-- Global rules -->
+	<record model="ir.rule.group" id="rule_group_contract_companies">
+            <field name="name">User in companies</field>
+            <field name="model" search="[('model', '=', 'contract')]"/>
+            <field name="global_p" eval="True"/>
+        </record>
+        <record model="ir.rule" id="rule_contract_companies">
+            <field name="domain"
+                eval="[('company', 'in', Eval('companies', []))]"
+                pyson="1"/>
+            <field name="rule_group" ref="rule_group_contract_companies"/>
+        </record>
+
+	<record model="ir.rule.group" id="rule_group_contract_review_companies">
+            <field name="name">User in companies</field>
+            <field name="model" search="[('model', '=', 'contract.review')]"/>
+            <field name="global_p" eval="True"/>
+        </record>
+        <record model="ir.rule" id="rule_contract_review_companies">
+            <field name="domain"
+                eval="[('company', 'in', Eval('companies', []))]"
+                pyson="1"/>
+            <field name="rule_group" ref="rule_group_contract_review_companies"/>
+        </record>
+
+	<!-- Menus -->
```

Comment:
I miss rules for contract.line and contract.consumption models

Reply:
I was strictly adhering to the MultiCompanyTest test which doesn't suggest their existence. Of course I can add them as well.

Reply:
BTW neither contract.line nor consumption have company fields. I think that is why the test does not list them and why rules 'User in companies' do not make sense for them.